tests/testthat/test-stroke.R

```diff
@@ -129,22 +129,23 @@ test_that("edges are not split if flow_mode is true", {
 })
 
 test_that("strokes can be formed starting from a given edge", {
-  skip(message = "stroke from edge to be implemented")
-  l1 <- sf::st_linestring(c(p1, p2, p3))
-  sfc <- sf::st_sfc(l1, l4, l7)
+  new_l1 <- sf::st_linestring(c(p1, p2, p3))
+  sfc <- sf::st_sfc(new_l1, l4, l7)
+  # p1 - p2 - p3
+  #         \
+  #           p5 - p6
   expected <- sf::st_sfc(sf::st_linestring(c(p1, p2, p5, p6)))
-  actual <- stroke(sfc, flow_mode = FALSE, from_edge = 3)
+  actual <- stroke(sfc, flow_mode = FALSE, from_edge = list(3))
   expect_setequal(actual, expected)
 })
 
-test_that("strokes can be formed starting from a given line segment", {
-  skip(message = "stroke from edge to be implemented")
-  l1 <- sf::st_linestring(c(p1, p2, p3))
-  l2 <- sf::st_linestring(c(p2, p5, p6))
-  sfc <- sf::st_sfc(l1, l2)
-  expected <- sf::st_sfc(sf::st_linestring(c(p1, p2, p5, p6)))
-  actual <- stroke(sfc, flow_mode = FALSE,
-                   from_edge = sf::st_linestring(c(p5, p6)))
+test_that("strokes can be formed starting from a given a list of edge ids", {
+  new_l1 <- sf::st_linestring(c(p1, p2, p3))
+  sfc <- sf::st_sfc(new_l1, l4, l7)
+  stroke_1 <- sf::st_linestring(c(p1, p2, p3))
+  stroke_2 <- sf::st_linestring(c(p1, p2, p5, p6))
+  expected <- sf::st_sfc(stroke_1, stroke_2)
+  actual <- stroke(sfc, flow_mode = FALSE, from_edge = list(1, 3))
```

Comment:
nice, I implemented your suggested test. However, the second stroke is also reversed i.e. it is sf::st_linestring(c(p6, p5, p2, p1)) now , is this expected? :thinking:

Reply:
Thank you! 

> nice, I implemented your suggested test. However, the second stroke is also reversed i.e. it is sf::st_linestring(c(p6, p5, p2, p1)) now , is this expected? 🤔

Oh good point, I guess it is! Since edge 2 has now p5 as start and p2 as end, it also grows in the opposite direction. Great the test picks this up. :)